internal/kafka/command_topic_produce.go

```diff
@@ -231,6 +235,31 @@ func serializeMessage(keyMetaInfo, valueMetaInfo []byte, data, delimiter string,
 	return append(keyMetaInfo, serializedKey...), append(valueMetaInfo, serializedValue...), nil
 }
 
+func getKeyAndValue(schemaBased bool, data, delimiter string) (string, string, error) {
+	dataSplit := strings.Split(data, delimiter)
+	if len(dataSplit) < 2 {
+		return "", "", errors.New(missingKeyOrValueErrorMsg)
+	}
+
+	if !schemaBased {
+		return strings.TrimSpace(dataSplit[0]), strings.TrimSpace(strings.Join(dataSplit[1:], delimiter)), nil
+	}
+
+	key := dataSplit[0]
+	if json.Valid([]byte(strings.TrimSpace(key))) {
+		return strings.TrimSpace(key), strings.TrimSpace(strings.Join(dataSplit[1:], delimiter)), nil
+	}
+
+	for i, substr := range dataSplit[1:] {
+		key += delimiter + substr
+		if json.Valid([]byte(strings.TrimSpace(key))) {
+			return strings.TrimSpace(key), strings.TrimSpace(strings.Join(dataSplit[i+2:], delimiter)), nil
```

Comment:
Is it possible to get an out of bounds panic here if `dataSplit` is too short?

Reply:
No. As long as `i+2 <= len(dataSplit)`, `dataSplit[i+2:]` is valid.

Since `i` ranges from `0` to `len(dataSplit[1:]) - 1 = len(dataSplit) - 2`, `i+2` is within bounds.

If `dataSplit[1:]` is empty (which shouldn't happen because there's a check at the top enforcing `len(dataSplit) >= 2`), then this is still fine because the loop doesn't run at all.